azurerm/data_factory.go

```diff
@@ -0,0 +1,74 @@
+package azurerm
+
+import (
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/datafactory/mgmt/2018-06-01/datafactory"
+)
+
+func expandDataFactoryParameters(input map[string]interface{}) map[string]*datafactory.ParameterSpecification {
+	output := make(map[string]*datafactory.ParameterSpecification)
+
+	for k, v := range input {
+		output[k] = &datafactory.ParameterSpecification{
+			Type:         datafactory.ParameterTypeString,
+			DefaultValue: v.(string),
+		}
+	}
+
+	return output
+}
+
+func expandDataFactoryAdditionalProperties(input map[string]interface{}) map[string]interface{} {
+	output := make(map[string]interface{})
+
+	for k, v := range input {
+		output[k] = v
+	}
+
+	return output
+}
+
+func flattenDataFactoryParameters(input map[string]*datafactory.ParameterSpecification) map[string]interface{} {
+	output := make(map[string]interface{})
+
+	for k, v := range input {
+		if v != nil {
+			// we only support string parameters at this time
+			val, ok := v.DefaultValue.(string)
+			if !ok {
+				log.Printf("[DEBUG] Skipping parameter %q since it's not a string", k)
+			}
+
+			output[k] = val
+		}
+	}
+
+	return output
+}
+
+func expandDataFactoryAnnotations(input []interface{}) *[]interface{} {
+	annotations := make([]interface{}, 0)
+
+	for _, annotation := range input {
+		annotations = append(annotations, annotation.(string))
+	}
+
+	return &annotations
+}
+
+func flattenDataFactoryAnnotations(input *[]interface{}) []string {
```

Comment:
I think you could use `utils.FlattenStringArray` instead of this?

Reply:
Weirdly enough, I think it'd be a call to utils.ExpandStringArray

Reply:
On second thought. I think it's needed because it's an *[]interface{} and we'd need to do some finagling to get it to a []string

Reply:
Ahh yea sorry, its so close to the signature, and as much as i'd want to ask if terraform would accept the interface array i know we should check they are all strings.

Reply:
No worries!